src/Tasks/Microsoft.NET.Build.Tasks/ProjectContext.cs

```diff
@@ -65,14 +65,14 @@ public IEnumerable<LockFileTargetLibrary> GetRuntimeLibraries(IEnumerable<string
                 allExclusionList.UnionWith(_lockFileTarget.GetPlatformExclusionList(PlatformLibrary, libraryLookup));
             }
 
-            if (privateAssetPackageIds?.Any() == true)
+            if (excludeFromPublishPackageIds?.Any() == true)
```

Comment:
I think that's actually needed because the value you're comparing to is `Nullable<bool>`